doc/dox_comments/header_files/aes.h

```diff
@@ -1974,3 +1974,1678 @@ int wc_AesCtsDecryptUpdate(Aes* aes, byte* out, word32* outSz,
+    \brief This function finalizes streaming AES XTS decryption. It
+    processes any remaining data and completes the decryption operation.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If parameters are invalid.
+
+    \param aes pointer to the XtsAes structure
+    \param out pointer to buffer to store final decrypted data
+    \param in pointer to final ciphertext data to decrypt
+    \param sz length of final data in bytes
+    \param stream pointer to XtsAesStreamData structure for streaming state
+
+    _Example_
+    \code
+    XtsAes aes;
+    struct XtsAesStreamData stream;
+    byte key[32] = { /* 256-bit key */ };
+    byte tweak[16] = { /* tweak value */ };
+    byte ciphertext[50] = { /* final encrypted data */ };
+    byte plaintext[50];
+    
+    wc_AesXtsSetKey(&aes, key, 32, AES_DECRYPTION, NULL, INVALID_DEVID);
+    wc_AesXtsDecryptInit(&aes, tweak, 16, &stream);
+    // ... update calls ...
+    int ret = wc_AesXtsDecryptFinal(&aes, plaintext, ciphertext, 50,
+                                    &stream);
+    if (ret != 0) {
+        // finalization failed
+    }
+    wc_AesXtsFree(&aes);
+    \endcode
+
+    \sa wc_AesXtsDecryptUpdate
+    \sa wc_AesXtsDecryptInit
+*/
+int wc_AesXtsDecryptFinal(XtsAes* aes, byte* out, const byte* in,
+                          word32 sz, struct XtsAesStreamData *stream);
+
+/*!
+    \ingroup AES
+    \brief This function retrieves the key size from an initialized AES
+    structure. It returns the size of the key currently set in the AES
+    object.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If aes or keySize is NULL.
+
+    \param aes pointer to the AES structure
+    \param keySize pointer to word32 to store the key size in bytes
+
+    _Example_
+    \code
+    Aes aes;
+    byte key[16] = { /* 128-bit key */ };
+    word32 keySize;
+    
+    wc_AesInit(&aes, NULL, INVALID_DEVID);
+    wc_AesSetKey(&aes, key, 16, NULL, AES_ENCRYPTION);
+    int ret = wc_AesGetKeySize(&aes, &keySize);
+    if (ret == 0) {
+        // keySize now contains 16
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesSetKey
+    \sa wc_AesInit
+*/
+int wc_AesGetKeySize(Aes* aes, word32* keySize);
+
+/*!
+    \ingroup AES
+    \brief This function initializes an AES structure with an ID. This is
+    useful for tracking or identifying specific AES instances in
+    applications that manage multiple AES contexts.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If aes or id is NULL, or if len is invalid.
+
+    \param aes pointer to the AES structure to initialize
+    \param id pointer to the ID buffer
+    \param len length of the ID in bytes
+    \param heap pointer to heap hint for memory allocation (can be NULL)
+    \param devId device ID for hardware acceleration (use INVALID_DEVID
+    for software)
+
+    _Example_
+    \code
+    Aes aes;
+    byte id[8] = { /* unique identifier */ };
+    
+    int ret = wc_AesInit_Id(&aes, id, 8, NULL, INVALID_DEVID);
```

Comment:
The _Id and _Label init helpers are only available with `WOLF_PRIVATE_KEY_ID` which is set for PKCS11 support. Please add comment indicating this